thread/linux/omrthreadnuma.c

```diff
@@ -134,42 +134,69 @@ initializeNumaNodeData(omrthread_library_t threadLibrary, uintptr_t numNodes)
 	} else {
 		DIR *nodes = NULL;
 		unsigned long nodeIndex = 0;
+		const char NODE_PATH[] = "/sys/devices/system/node/";
 		for (nodeIndex = 0; nodeIndex <= numNodes; nodeIndex++) {
 			CPU_ZERO(&numaNodeData[nodeIndex].cpu_set);
 			numaNodeData[nodeIndex].cpu_count = 0;
 		}
 
-		nodes = opendir("/sys/devices/system/node/");
+		nodes = opendir(NODE_PATH);
 		if (NULL == nodes) {
 			result = -1;
 		} else {
 			struct dirent *node = readdir(nodes);
-			while (NULL != node) {
-				if (1 == sscanf(node->d_name, "node%lu", &nodeIndex)) {
-					if (nodeIndex < numNodes) {
-						DIR *cpus = NULL;
-						char nodeName[sizeof("/sys/devices/system/node/") + NAME_MAX + 1] = "/sys/devices/system/node/";
-						strcat(nodeName, node->d_name);
-						cpus = opendir(nodeName);
-						if (NULL != cpus) {
-							struct dirent *cpu = readdir(cpus);
-							while (NULL != cpu) {
-								unsigned long cpuIndex = 0;
-								if (1 == sscanf(cpu->d_name, "cpu%lu", &cpuIndex)) {
-									CPU_SET(cpuIndex, &numaNodeData[nodeIndex + 1].cpu_set);
-									numaNodeData[nodeIndex + 1].cpu_count += 1;
-
-									/* add all cpus to the synthetic 0 node */
-									CPU_SET(cpuIndex, &numaNodeData[0].cpu_set);
-									numaNodeData[0].cpu_count += 1;
+			if (NULL != node) {
+				const char CPULIST[] = "/cpulist";
+				char pathBuffer[sizeof(NODE_PATH) + sizeof(CPULIST) + 16]; /* extra space for "node<n>" & terminating null */
+				strcpy(pathBuffer, NODE_PATH);
+				char *nodeRoot = pathBuffer + sizeof(NODE_PATH) - 1; /* shortcut to the end of the string */
+				do {
+					if (1 == sscanf(node->d_name, "node%lu", &nodeIndex)) {
+						if (nodeIndex < numNodes) {
+							DIR *cpus = NULL;
+							strcpy(nodeRoot, node->d_name);
+							strcat(nodeRoot, CPULIST);
+							FILE *cpulistFile = fopen(pathBuffer, "r");
+							if (NULL != cpulistFile) {
+								int numFound = 0;
+								do {
+									unsigned long cpuIndex = 0;
+									numFound = fscanf(cpulistFile, "%lu", &cpuIndex);
+									if (1 == numFound) {
```

Comment:
Can you add a description of the /cpulist file format?
Looking at the info on my of local machines, I see:
```
$ cat /sys/devices/system/node/node0/cpulist
0-15
```
How did you validate the same CPUs are set using both the new and old way?